holopy/inference/cmaes.py

```diff
@@ -80,25 +81,29 @@ def __init__(self, npixels=None, resample_pixels=True,
             def weight_function(x, n):
                 return (x + 1) <= (parent_fraction * n)
         self.weights = weight_function
+        self.walker_initial_pos = walker_initial_pos
         self.tols = {'maxiter':2000, 'tolx':0.001, 'tolfun':0.1,
                      'tolstagnation':100}
         self.tols.update(tols)
         self.seed = seed
         self.parallel = parallel
 
-    def optimize(self, model, data, popsize=None, walker_initial_pos=None):
+    def fit(self, model, data):
         parameters = model._parameters
         time_start = time.time()
         if self.npixels is not None and self.new_pixels is None:
             data = make_subset_data(data, pixels=self.npixels, seed=self.seed)
-        if popsize is None:
-            numpars = len(parameters)
-            popsize = int(2 + numpars + np.sqrt(numpars)) #cma uses 4+3*ln(n)
-        if walker_initial_pos is None:
-            walker_initial_pos = model.generate_guess(popsize, seed=self.seed)
+        if self.popsize is None:
+            npars = len(parameters)
+            self.popsize = int(2 + npars + np.sqrt(numpars))#cma uses 4+3*ln(n)
```

Comment:
This is a typo, I believe. `npars` is defined but now `numpars` is not.

If this is the case, can we capture this failure with a unit test before fixing it?

Reply:
done. There are embarrasingly few tests for the cma.py module, but I added this one!